### PR TITLE
Use project name for dataset.json downloads

### PR DESCRIPTION
**Pull request by microbit-grace**
Rename dataset.json -> project name-data-samples.json

https://microbit-global.monday.com/boards/1550536443/pulses/1657153875 (Private)

### CODE REVIEW

File: src/store.ts [hunk 1]
```diff
@@ -408,14 +408,17 @@ const createMlStore = (logging: Logging) => {
...
             const a = document.createElement("a");
             a.setAttribute(
               "href",
               "data:application/json;charset=utf-8," +
                 encodeURIComponent(JSON.stringify(gestures, null, 2))
             );
-            a.setAttribute("download", "dataset");
+            a.setAttribute(
+              "download",
+              `${project.header?.name || "Untitled"}-data-samples`
```
Comment (microbit-grace) [inline src/store.ts:420]: Not sure if we need to process the project name for json filename as well (e.g. strip out the spaces/punctuations) 🤔

**Comment by microbit-matt-hillsdon** [comment]
This works as expected once the project name is set (e.g. via Save) but needs the change to prompt for it if needed as per save flow. I suspect that's why it's draft but just commenting seeing as a took a peek 😄

**Comment by microbit-grace** [comment]
> This works as expected once the project name is set (e.g. via Save) but needs the change to prompt for it if needed as per save flow. I suspect that's why it's draft but just commenting seeing as a took a peek 😄

Ahh didn't realise it needed similar flow to save flow for changing name! Thanks for the comment and clarifying 😁

**Review by microbit-matt-hillsdon — APPROVED** [review]
Working well, thanks.

**Pull request merged**